File: vane-core/src/main/java/org/oddlama/vane/util/Conversions.java
package org.oddlama.vane.util;

public class Conversions {

    public static long msToTicks(long ms) {
        return ms / 50;
    }

    public static long ticksToMs(long ticks) {
        return ticks * 50;
    }

    public static int expForLevel(int level) {
        if (level < 17) {
            return level * level + 6 * level;
        } else if (level < 32) {
            return (int) (2.5 * level * level - 40.5 * level) + 360;
        } else {
            return (int) (4.5 * level * level - 162.5 * level) + 2220;
        }
    }
}

File: vane-core/src/main/kotlin/org/oddlama/vane/util/Conversions.kt
package org.oddlama.vane.util

/**
 * Converts milliseconds to server ticks.
 */
fun msToTicks(ms: Long): Long = ms / 50

/**
 * Converts server ticks to milliseconds.
 */
fun ticksToMs(ticks: Long): Long = ticks * 50

/**
 * Returns cumulative experience required for the given level.
 */
fun expForLevel(level: Int): Int = when {
    level < 17 -> level * level + 6 * level
    level < 32 -> (2.5 * level * level - 40.5 * level).toInt() + 360
    else       -> (4.5 * level * level - 162.5 * level).toInt() + 2220
}
